File: tests/python/pipeline_util/test-load.py
#!/usr/bin/env python
#ckwg +5
# Copyright 2011 by Kitware, Inc. All Rights Reserved. Please refer to
# KITWARE_LICENSE.TXT for licensing information, or contact General Counsel,
# Kitware, Inc., 28 Corporate Drive, Clifton Park, NY 12065.


def log(msg):
    import sys
    sys.stderr.write("%s\n" % msg)


def test_import():
    try:
        import vistk.pipeline_util.load
    except:
        log("Error: Failed to import the load module")


def test_create():
    from vistk.pipeline_util import load

    load.Token()
    load.ConfigFlag()
    load.ConfigFlags()
    load.ConfigProvider()
    load.ConfigKeyOptions()
    load.ConfigKey()
    load.ConfigValue()
    load.ConfigValues()
    load.MapOptions()
    load.InputMap()
    load.InputMaps()
    load.OutputMap()
    load.OutputMaps()
    load.ConfigBlock()
    load.ProcessBlock()
    load.ConnectBlock()
    load.GroupBlock()
    load.PipeBlock()
    load.PipeBlocks()


def test_api_calls():
    from vistk.pipeline import config
    from vistk.pipeline import process
    from vistk.pipeline import process_registry
    from vistk.pipeline_util import load

    o = load.ConfigKeyOptions()
    o.flags
    o.provider
    o.flags = load.ConfigFlags()
    o.provider = load.ConfigProvider()

    o = load.ConfigKey()
    o.key_path
    o.options
    o.key_path = config.ConfigKeys()
    o.options = load.ConfigKeyOptions()

    o = load.ConfigValue()
    o.key
    o.value
    o.key = load.ConfigKey()
    o.value = config.ConfigValue()

    o = load.MapOptions()
    o.flags
    o.flags = process.PortFlags()

    o = load.InputMap()
    o.options
    o.from_
    o.to
    o.options = load.MapOptions()
    o.from_ = process.Port()
    o.to = process.PortAddr()

    o = load.OutputMap()
    o.options
    o.from_
    o.to
    o.options = load.MapOptions()
    o.from_ = process.PortAddr()
    o.to = process.Port()

    o = load.ConfigBlock()
    o.key
    o.values
    o.key = config.ConfigKeys()
    o.values = load.ConfigValues()

    o = load.ProcessBlock()
    o.name
    o.type
    o.config_values
    o.name = process.ProcessName()
    o.type = process_registry.ProcessType()
    o.config_values = load.ConfigValues()

    o = load.ConnectBlock()
    o.from_
    o.to
    o.from_ = process.PortAddr()
    o.to = process.PortAddr()

    o = load.GroupBlock()
    o.name
    o.config_values
    o.input_mappings
    o.output_mappings
    o.name = process.ProcessName()
    o.config_values = load.ConfigValues()
    o.input_mappings = load.InputMaps()
    o.output_mappings = load.OutputMaps()

    o = load.PipeBlock()
    o.config = load.ConfigBlock()
    o.config
    o.process = load.ProcessBlock()
    o.process
    o.connect = load.ConnectBlock()
    o.connect
    o.group = load.GroupBlock()
    o.group


def test_simple_pipeline(path):
    from vistk.pipeline_util import load

    blocks = load.load_pipe_file(path)
    with open(path, 'r') as fin:
        load.load_pipe(fin)


def main(testname, path):
    if testname == 'import':
        test_import()
    elif testname == 'create':
        test_create()
    elif testname == 'api_calls':
        test_api_calls()
    elif testname == 'simple_pipeline':
        test_simple_pipeline(path)
    else:
        log("Error: No such test '%s'" % testname)


if __name__ == '__main__':
    import os
    import sys

    if not len(sys.argv) == 5:
        log("Error: Expected four arguments")
        sys.exit(1)

    testname = sys.argv[1]

    os.chdir(sys.argv[2])

    sys.path.append(sys.argv[3])

    pipeline_dir = sys.argv[4]

    path = os.path.join(pipeline_dir, '%s.pipe' % testname)

    main(testname, path)

File: tests/lua/pipeline_util/test-load.lua
#!/usr/bin/env python
--ckwg +5
-- Copyright 2011 by Kitware, Inc. All Rights Reserved. Please refer to
-- KITWARE_LICENSE.TXT for licensing information, or contact General Counsel,
-- Kitware, Inc., 28 Corporate Drive, Clifton Park, NY 12065.


function log(msg)
    io.stderr:write(string.format("%s\n", msg))
end


function test_import()
    if not pcall(require, "vistk.pipeline_util.load") then
        log("Error: Failed to import the load module")
    end
end


function test_create()
    require("vistk.pipeline_util.load")

    vistk.pipeline_util.token()
    vistk.pipeline_util.config_flag()
    vistk.pipeline_util.config_flags()
    vistk.pipeline_util.config_provider()
    vistk.pipeline_util.config_key_options()
    vistk.pipeline_util.config_key()
    vistk.pipeline_util.config_value()
    vistk.pipeline_util.config_values()
    vistk.pipeline_util.map_options()
    vistk.pipeline_util.input_map()
    vistk.pipeline_util.input_maps()
    vistk.pipeline_util.output_map()
    vistk.pipeline_util.output_maps()
    vistk.pipeline_util.config_block()
    vistk.pipeline_util.process_block()
    vistk.pipeline_util.connect_block()
    vistk.pipeline_util.group_block()
    vistk.pipeline_util.pipe_block()
    vistk.pipeline_util.pipe_blocks()
end


function test_api_calls()
    require("vistk.pipeline.config")
    require("vistk.pipeline.process")
    require("vistk.pipeline.process_registry")
    require("vistk.pipeline_util.load")

    local tmp
    local o

    o = vistk.pipeline_util.config_key_options()
    tmp = o.flags
    tmp = o.provider
    o.flags = vistk.pipeline_util.config_flags()
    o.provider = vistk.pipeline_util.config_provider()

    o = vistk.pipeline_util.config_key()
    tmp = o.key_path
    tmp = o.options
    o.key_path = vistk.pipeline.config_keys()
    o.options = vistk.pipeline_util.config_key_options()

    o = vistk.pipeline_util.config_value()
    tmp = o.key
    tmp = o.value
    o.key = vistk.pipeline_util.config_key()
    o.value = vistk.pipeline.config_value()

    o = vistk.pipeline_util.map_options()
    tmp = o.flags
    o.flags = vistk.pipeline.port_flags()

    o = vistk.pipeline_util.input_map()
    tmp = o.options
    tmp = o.from
    tmp = o.to
    o.options = vistk.pipeline_util.map_options()
    o.from = vistk.pipeline.port()
    o.to = vistk.pipeline.port_addr()

    o = vistk.pipeline_util.output_map()
    tmp = o.options
    tmp = o.from
    tmp = o.to
    o.options = vistk.pipeline_util.map_options()
    o.from = vistk.pipeline.port_addr()
    o.to = vistk.pipeline.port()

    o = vistk.pipeline_util.config_block()
    tmp = o.key
    tmp = o.values
    o.key = vistk.pipeline.config_keys()
    o.values = vistk.pipeline_util.config_values()

    o = vistk.pipeline_util.process_block()
    tmp = o.name
    tmp = o.type
    tmp = o.config_values
    o.name = vistk.pipeline.process_name()
    o.type = process_registry.process_type()
    o.config_values = vistk.pipeline_util.config_values()

    o = vistk.pipeline_util.connect_block()
    tmp = o.from
    tmp = o.to
    o.from = vistk.pipeline.port_addr()
    o.to = vistk.pipeline.port_addr()

    o = vistk.pipeline_util.group_block()
    tmp = o.name
    tmp = o.config_values
    tmp = o.input_mappings
    tmp = o.output_mappings
    o.name = vistk.pipeline.process_name()
    o.config_values = vistk.pipeline_util.config_values()
    o.input_mappings = vistk.pipeline_util.input_maps()
    o.output_mappings = vistk.pipeline_util.output_maps()

    o = vistk.pipeline_util.pipe_block()
    o.config = vistk.pipeline_util.config_block()
    tmp = o.config
    o.process = vistk.pipeline_util.process_block()
    tmp = o.process
    o.connect = vistk.pipeline_util.connect_block()
    tmp = o.connect
    o.group = vistk.pipeline_util.group_block()
    tmp = o.group
end


function test_simple_pipeline(path)
    require("vistk.pipeline_util.load")

    local blocks = vistk.pipeline_util.load_pipe_file(path)
    local fin = io.open(path, 'r')
    vistk.pipeline_util.load_pipe(fin)
    fin:close()
end


function main(testname, path)
    if testname == 'import' then
        test_import()
    elseif testname == 'create' then
        test_create()
    elseif testname == 'api_calls' then
        test_api_calls()
    elseif testname == 'simple_pipeline' then
        test_simple_pipeline(path)
    else
        log(string.format("Error: No such test '%s'", testname))
    end
end


if #arg ~= 3 then
    log("Error: Expected three arguments")
    os.exit(1)
end

local testname = arg[1]

package.cpath = string.format("%s/?@CMAKE_SHARED_LIBRARY_SUFFIX@;%s", arg[2], package.cpath)

local pipeline_dir = arg[3]

local path = string.format("%s/%s.pipe", pipeline_dir, testname)

local status, err = pcall(main, testname, path)

if not status then
    log(string.format("Error: Unexpected exception: %s", err))
end
